code/go/0chain.net/smartcontract/minersc/fees.go

```diff
@@ -201,30 +201,31 @@ func (msc *MinerSmartContract) adjustViewChange(gn *GlobalNode,
 	}
 
 	var waited int
-	for k := range dmn.SimpleNodes {
-		if !dmn.Waited[k] {
-			delete(dmn.SimpleNodes, k)
+	for _, miner := range dmn.Nodes {
+		if !dmn.Waited[miner.Key] {
+			dmn.DeleteNode(miner.Key)
 			continue
 		}
 		waited++
 	}
 
 	if err := cstate.WithActivation(balances, "hermes",
 		func() error {
-			err = dmn.reduceNodes(true, gn, balances)
-			if err == nil && waited < dmn.K {
-				err = fmt.Errorf("< K miners succeed 'wait' phase: %d < %d",
-					waited, dmn.K)
-			}
-			if err != nil {
-				var prev = gn.prevMagicBlock(balances)
-				gn.MustUpdateBase(func(gnb *globalNodeBase) error {
-					gnb.ViewChange = prev.StartingRound
-					return nil
-				})
-				err = nil
-			}
-			return err
```

Comment:
is it safe to delete it? in theory new transactons after network partial update will break the mpt consistency, not sure though if it's practical to attack